### ISSUE TITLE
marker/shield fill color while using file

### ISSUE THREAD

**Issue by farfromrefug** [CartoDB/mobile-carto-libs#18]
Shouldn't the fill color colorize the marker/shield file ? 
It is happening with `polygon-pattern-fill`, even though not documented in carto css, so i would expect/like it to work too. What do you think?

**Comment by mtehver** [comment]
I believe the current implementation mirrors Mapnik behavior, marker-fill is used when drawing custom marker (ellipse or arrow). `polygon-pattern-fill` is a custom extension that is not supported by Mapnik at all.

**Comment by farfromrefug** [comment]
@mtehver yes i saw that. But in the "continuity" of `polygon-pattern-fill` which is dead useful, couldn't `marker-fill` do the same while using file? It would change the behavior, just add more features. The only edge case i would see is if someone had `marker-fill` set with `file` without actually knowing it (as it had no side effect).

**Comment by mtehver** [comment]
@farfromrefug I added 'marker-color' property that can be used for this. I analyzed different options and this seemed the best one. Note that 'marker-opacity' already existed and this worked both with file based and internal markers. So 'marker-color' is also applied to both file based and internal markers while 'marker-fill', 'marker-fill-opacity', 'marker-stroke', 'marker-stroke-opacity' are only used for internal markers.
